### PR TITLE
[EPMEDU-1998]: Reduce size of downloaded images

### PR DESCRIPTION
**Pull request by Fileek**
[Jira issue](https://jira.epam.com/jira/browse/EPMEDU-1998)

  - Prevented image duplication in cache by using disk cache keys
  - Moved `StorageApi` to a separate module networkstorage
  - Created `NetworkFile` wrapper

| Before | After |
| ------------- | ------------- |
| <video src="https://github.com/AnimealProject/animeal_android/assets/83027107/49c3819a-25f8-42bb-9bb4-0315afd764d8"> | <video src="https://github.com/AnimealProject/animeal_android/assets/83027107/96b54a82-e696-45e4-936e-a38f89515ce5"> |

### CODE REVIEW

File: shared/feature/feeding/src/main/java/com/epmedu/animeal/feeding/data/mapper/DataToDomainFeedingPointMapper.kt [hunk 1]
```diff
@@ -6,28 +6,31 @@ import com.epmedu.animeal.common.constants.DefaultConstants.EMPTY_STRING
...
-        isFavourite = isFavourite,
-        location = MapLocation(
-            latitude = location?.lat ?: 0.0,
-            longitude = location?.lon ?: 0.0
-        ),
-        images = images
-    )
\ No newline at end of file
+internal suspend fun DataFeedingPoint.toDomainFeedingPoint(
+    getImageFromName: suspend (String) -> NetworkFile?,
```
Comment (ozkarram) [inline shared/feature/feeding/src/main/java/com/epmedu/animeal/feeding/data/mapper/DataToDomainFeedingPointMapper.kt]: rename to `getImageFrom`

File: shared/feature/networkstorage/src/main/java/com/epmedu/animeal/networkstorage/data/api/StorageApi.kt [hunk 2]
```diff
@@ -7,7 +7,7 @@ interface StorageApi {
 
     suspend fun uploadFile(fileName: String, uri: Uri): ApiResult<Unit>
 
-    suspend fun parseAmplifyUrl(imageId: String): String
+    suspend fun getUrlFromFileName(fileName: String): String
```
Comment (ozkarram) [inline shared/feature/networkstorage/src/main/java/com/epmedu/animeal/networkstorage/data/api/StorageApi.kt]: `getUrlFrom`

**Review by ozkarram — CHANGES_REQUESTED** [review]
Just some naming changes

File: shared/feature/feeding/src/main/java/com/epmedu/animeal/feeding/data/mapper/DataToDomainFeedingPointMapper.kt (hunk 1 above)
Comment (Fileek) [reply shared/feature/feeding/src/main/java/com/epmedu/animeal/feeding/data/mapper/DataToDomainFeedingPointMapper.kt]: Done

File: shared/feature/networkstorage/src/main/java/com/epmedu/animeal/networkstorage/data/api/StorageApi.kt (hunk 2 above)
Comment (Fileek) [reply shared/feature/networkstorage/src/main/java/com/epmedu/animeal/networkstorage/data/api/StorageApi.kt]: Done

**Review by ozkarram — APPROVED** [review]

**Pull request merged**